Microsoft.Toolkit.Uwp.UI.Controls/ImageCropper/ImageCropper.Extensions.cs

```diff
@@ -0,0 +1,288 @@
+            WriteableBitmap croppedBitmap;
+            var sourceStream = writeableBitmap.PixelBuffer.AsStream();
+            var buffer = new byte[sourceStream.Length];
+            await sourceStream.ReadAsync(buffer, 0, buffer.Length);
+            using (var memoryRandom = new InMemoryRandomAccessStream())
+            {
+                var encoder = await BitmapEncoder.CreateAsync(BitmapEncoder.PngEncoderId, memoryRandom);
+                encoder.SetPixelData(BitmapPixelFormat.Bgra8, BitmapAlphaMode.Premultiplied, (uint)writeableBitmap.PixelWidth, (uint)writeableBitmap.PixelHeight, 96.0, 96.0, buffer);
+                encoder.BitmapTransform.Bounds = new BitmapBounds
+                {
+                    X = x,
+                    Y = y,
+                    Height = height,
+                    Width = width
+                };
+                await encoder.FlushAsync();
+                croppedBitmap = new WriteableBitmap(
+                    (int)encoder.BitmapTransform.Bounds.Width,
+                    (int)encoder.BitmapTransform.Bounds.Height);
+                croppedBitmap.SetSource(memoryRandom);
+            }
+
+            return croppedBitmap;
+        }
+
+        /// <summary>
+        /// Save the cropped image to a file.
+        /// </summary>
+        /// <param name="writeableBitmap">The source image.</param>
+        /// <param name="imageFile">The target file.</param>
+        /// <param name="encoderId">The encoderId of BitmapEncoder</param>
+        /// <returns>Task</returns>
+        internal static async Task RenderToFile(WriteableBitmap writeableBitmap, StorageFile imageFile, Guid encoderId)
+        {
+            using (var stream = await imageFile.OpenAsync(FileAccessMode.ReadWrite, StorageOpenOptions.None))
+            {
+                var encoder = await BitmapEncoder.CreateAsync(encoderId, stream);
+                var pixelStream = writeableBitmap.PixelBuffer.AsStream();
+                var pixels = new byte[pixelStream.Length];
+                await pixelStream.ReadAsync(pixels, 0, pixels.Length);
+                encoder.SetPixelData(
+                    BitmapPixelFormat.Bgra8,
+                    BitmapAlphaMode.Premultiplied,
+                    (uint)writeableBitmap.PixelWidth,
+                    (uint)writeableBitmap.PixelHeight,
+                    96.0,
+                    96.0,
+                    pixels);
+                await encoder.FlushAsync();
+            }
+        }
+
+        /// <summary>
+        /// Gets the closest point in the rectangle to a given point.
+        /// </summary>
+        /// <param name="targetRect">The rectangle.</param>
+        /// <param name="point">The test point.</param>
+        /// <returns>A point within a rectangle.</returns>
+        internal static Point GetSafePoint(Rect targetRect, Point point)
+        {
+            var safePoint = new Point(point.X, point.Y);
+            if (safePoint.X < targetRect.X)
```

Comment:
safePoint.X=Math.Max(safePoint.X, targetRect.X)

Reply:
That's fine, but still judging `if (safePoint.X > targetRect.X + targetRect.Width)`, the code seems to be getting harder to understand.

Reply:
We could combine the statements and just add a comment:

```
// Constrain point within bounds
safePoint.X = Math.Min(Math.Max(safePoint.X, targetRect.X), targetRect.X + targetRect.Width);
```

Thought we had a constrain extension, but maybe that was in one of my other projects...